releaseNotes/SpringCloudAzureAppConfigurationConfig.md

```diff
@@ -2,6 +2,47 @@
 
 [Source code][source_code] | [Package (Maven)][package] | [Product documentation][docs] | [Samples][samples]
 
+## 1.1.1/1.2.1 - January 13, 2020
+
```

Comment:
We should use full sentences and publicly known wording (instead of acronyms) for the release notes wherever we can. Whenever possible, we should also explain why we are making each change. For example, instead of this
- Watch enabled has been removed. To control the refresh rate the cache expiration configuration can be used. By default 30s.

We can have something like
- The settings below are removed as the timer-based configuration watch has been deprecated.
  ```properties
  spring.cloud.azure.appconfiguration.watch.enabled
  spring.cloud.azure.appconfiguration.watch.delay
  ```
- The configuration is refreshed based on application activities.
    - The configuration is cached and the default cache expiration time is 30 seconds. The configuration won't be refreshed until the cache expiration time window is reached. This value can be modified, for example,
  ```properties
  spring.cloud.azure.appconfiguration.cache-expiration = 60
  ```
    - In a web application, this library will signal configuration refresh automatically as long as there are incoming requests to the application (quick question, will this cause our library to depend on some web library?).
    - In a non-web application, the user needs to xxx (call `AzureCloudConfigRefresh`? is there a code snippet?) to signal configuration refresh at places where application activities occur.

There are also duplicated information. Please clean them up.

Reply:
@zhenlan When this was initially changed we already had a dependency on org.springframework.web, from the shared dependency of the spring-cloud-azure-autoconfigure library. But now with use of the Azure sdk, that dependency was removed, so now this is the only thing causing us to have a web dependency. We can rethink it, we could either remove it completely, or drop it from listening to `ServletRequestHandledEvent` to `ApplicationEvent`, but that may be to often. The only other option would be to have the listener be in the sample and have the user implement it.

Reply:
In this case, I think we should create a separate library for the web.